client.go

```diff
@@ -315,6 +315,15 @@ func (uc *Client) isEnabled(feature string, options ...FeatureOption) (enabled b
 //
 // It is safe to call this method from multiple goroutines concurrently.
 func (uc *Client) GetVariant(feature string, options ...VariantOption) *api.Variant {
+	variant := uc.GetVariantWithoutMetrics(feature, options...)
+	defer func() {
+		uc.metrics.countVariants(feature, variant.Enabled, variant.Name)
+	}()
+	return variant
+}
+
+// GetVariantWithoutMetrics abstracts away the logic for resolving a variant without metrics
+func (uc *Client) GetVariantWithoutMetrics(feature string, options ...VariantOption) *api.Variant {
```

Comment:
I think this should be lowercase, we don't want to expose this through the public API (by default uppercase method names are public in go, and lowercase are private).

Reply:
Thanks! I didn't know that! 

@nunogois learning opportunity here :) https://go.dev/doc/effective_go#names

Reply:
[9941bc6](https://github.com/Unleash/unleash-client-go/pull/124/commits/9941bc664d535df08fd8b32a28966a73c07eaee4)